### PR TITLE
Delete Observation Button added in overflow menu

### LINKED ISSUE #70: [Feature panel] Delete observation flow
Only record creator and project managers should be allowed to delete records. 

@hintzemichael should this be added to the overflow menu on records in the side panel?

### PR DESCRIPTION
**Pull request by gauravchalana1**
Hey @gino-m Please review the PR for the Fix #70 and share your comments :) 

image
image

### CODE REVIEW

File: web-ng/src/app/components/feature-panel/feature-panel.component.html [hunk 1]
```diff
@@ -52,6 +52,13 @@
           <mat-icon>edit</mat-icon>
           <span>Edit observation</span>
         </button>
+        <button
+          mat-menu-item
+          (click)="onDeleteObservationClick(observation.id)"
+        >
+          <mat-icon>delete_forever</mat-icon>
+          <span>Delete</span>
```
Comment (gino-m) [inline web-ng/src/app/components/feature-panel/feature-panel.component.html:60]: Let's use "Delete observation" for consistency and clarity.

File: web-ng/src/app/components/feature-panel/feature-panel.component.ts [hunk 2]
```diff
@@ -20,31 +20,42 @@ import { FeatureService } from './../../services/feature/feature.service';
...
 // TODO: Rename "FeatureDetailsComponent".
 @Component({
   selector: 'ground-feature-panel',
   templateUrl: './feature-panel.component.html',
   styleUrls: ['./feature-panel.component.css'],
 })
-export class FeaturePanelComponent {
+export class FeaturePanelComponent implements OnInit, OnDestroy {
+  projectId?: string | null;
+  observationId?: string | null;
```
Comment (gino-m) [inline web-ng/src/app/components/feature-panel/feature-panel.component.ts:42]: Can `| null` be removed from these two, letting `?` (undefined) cover the 'missing' case?

File: web-ng/src/app/components/feature-panel/feature-panel.component.ts [hunk 3]
```diff
@@ -72,6 +83,20 @@ export class FeaturePanelComponent {
       );
   }
 
+  ngOnInit() {
+    this.subscription.add(
+      this.navigationService.getProjectId$().subscribe(id => {
+        this.projectId = id;
```
Comment (gino-m) [inline web-ng/src/app/components/feature-panel/feature-panel.component.ts:89]: I think you can convert from `string | undefined | null` to `string | undefined` by replacing `id` with `id || undefined`.

File: web-ng/src/app/components/feature-panel/feature-panel.component.ts [hunk 4]
```diff
@@ -72,6 +83,20 @@ export class FeaturePanelComponent {
...
+      })
+    );
+
+    this.subscription.add(
+      this.navigationService.getObservationId$().subscribe(id => {
+        this.observationId = id;
```
Comment (gino-m) [inline web-ng/src/app/components/feature-panel/feature-panel.component.ts:95]: Same here.

**Review by gino-m — CHANGES_REQUESTED** [review]
👍 Great fix! A few minor suggestions, otherwise LGTM!

**Comment by gauravchalana1** [comment]
Hey @gino-m , please re-review the changes and share your comments :) tyty

**Review by gino-m — APPROVED** [review]
Nicely done! LGTM 🚀

**Pull request merged**